inference-engine/src/transformations/src/transformations/convert_extract_image_patches_to_reorg_yolo.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/convert_extract_image_patches_to_reorg_yolo.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/opsets/opset3.hpp>
+#include <ngraph/rt_info.hpp>
+
+void ngraph::pass::ConvertExtractImagePatchesToReorgYolo::convert_extract_image_patches_to_reorg_yolo() {
+    auto image = std::make_shared<pattern::op::Label>(element::f32, Shape{1, 1, 1, 1});
+    auto extract_image_patches = std::make_shared<ngraph::opset3::ExtractImagePatches>(image, Shape{1, 1}, Strides{1, 1}, Shape{1, 1},
+            ngraph::op::PadType::VALID);
+
+    ngraph::graph_rewrite_callback callback = [](pattern::Matcher& m) {
+        auto extract_image_patches = std::dynamic_pointer_cast<ngraph::opset3::ExtractImagePatches>(m.get_match_root());
+
+        if (!extract_image_patches) {
+            return false;
+        }
+
+        if (extract_image_patches->get_strides() != extract_image_patches->get_sizes()) {
+            return false;
+        }
+
+        auto p_shape_input = extract_image_patches->get_input_partial_shape(0);
+        auto strides = extract_image_patches->get_strides();
+
+        // Check that ExtractImagePatches input have static shape
+        if (p_shape_input[2].is_dynamic() && p_shape_input[3].is_dynamic()) {
```

Comment:
Should be `||` instead of `&&`

Reply:
Also, lets be paranoid - add a check that `p_shape_input` has at least 4 elements so we don't try to use elements outside of the vector.